components/brave_rewards/browser/database/database_promotion.cc

```diff
@@ -284,4 +285,25 @@ ledger::PromotionMap DatabasePromotion::GetAllRecords(sql::Database* db) {
   return map;
 }
 
+bool DatabasePromotion::DeleteRecordList(
+      sql::Database* db,
+      const std::vector<std::string>& id_list) {
+  if (id_list.size() == 0) {
```

Comment:
Using `size() == 0` to check if a container is empty is a bad habit. 
If later someone changes `std::vector` to other non-block allocated container, the complexity for this call will change from constant to linear. There is an `empty()` function for this purpose. It returns a boolean.

Reply:
fixed

Reply:
I'm just passing by - all STL containers either have constant complexity `size()`, or do not have this method at all (like `forward_list`).

Reply:
I was thinking about `stl::list`.  But yes, from C++11 the `list::size` has a constant complexity too.  Even in  this case `empty` is better then retrieving a number of items, comparing it to 0  and get an answer.  Good catch, thanks.